src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubConfiguration.cs

```diff
@@ -69,6 +69,11 @@ public EventHubConfiguration()
         }
 
         /// <summary>
+        /// Sets the frequency at which to checkpoint event batches.
+        /// </summary>
+        public int BatchCheckpointFrequency { get; set; } = 1;
```

Comment:
Yep - single dispatch will pull a batch -> spawn batchSize # of tasks -> checkpoint after all batches complete